samza-core/src/main/scala/org/apache/samza/coordinator/JobModelManager.scala

```diff
@@ -128,10 +132,7 @@ object JobModelManager extends Logging {
       taskName -> Integer.valueOf(taskModel.getChangelogPartition.getPartitionId)
     }}.toMap ++ previousChangelogPartitionMapping.asScala
     info("Saving task-to-changelog partition mapping: %s" format newChangelogPartitionMapping)
-    changelogManager.writeChangeLogPartitionMapping(newChangelogPartitionMapping.asJava)
-
-    createChangeLogStreams(config, jobModel.maxChangeLogStreamPartitions)
-    createAccessLogStreams(config, jobModel.maxChangeLogStreamPartitions)
+    jobModel.setChangelogTaskPartitionMappings(newChangelogPartitionMapping.asJava);
```

Comment:
I debated both sides because the JobModel defines the task to partition, but it is cleaner like you said to completely decouple changelog from JobModel. Will look into your suggestion.